background.js

```diff
@@ -183,71 +187,53 @@ function videoIDChange(currentVideoID, tabId) {
     id: currentVideoID
   });
 
-  //not a url change
-  if (sponsorVideoID == currentVideoID){
-    return;
-  }
-  sponsorVideoID = currentVideoID;
-
-  //warn them if they had unsubmitted times
-  if (previousVideoID != null) {
-    //get the sponsor times from storage
-    let sponsorTimeKey = 'sponsorTimes' + previousVideoID;
-    chrome.storage.sync.get([sponsorTimeKey], function(result) {
-      let sponsorTimes = result[sponsorTimeKey];
-
-      if (sponsorTimes != undefined && sponsorTimes.length > 0) {
-        //warn them that they have unsubmitted sponsor times
-        chrome.notifications.create("stillThere" + Math.random(), {
-          type: "basic",
-          title: "Do you want to submit the sponsor times for watch?v=" + previousVideoID + "?",
-          message: "You seem to have left some sponsor times unsubmitted. Go back to that page to submit them (they are not deleted).",
-          iconUrl: "./icons/LogoSponsorBlocker256px.png"
-        });
-      }
+  chrome.storage.sync.get(["sponsorVideoID", "previousVideoID"], function(result) {
+    const sponsorVideoID = result.sponsorVideoID;
+    const previousVideoID = result.previousVideoID;
 
-      //set the previous video id to the currentID
-      previousVideoID = currentVideoID;
+    //not a url change
+    if (sponsorVideoID == currentVideoID){
+      return;
+    }
+
+    chrome.storage.sync.set({
+      "sponsorVideoID": currentVideoID
     });
-  } else {
-    previousVideoID = currentVideoID;
-  }
-}
 
-function getUserID(callback) {
```

Comment:
I deleted `getUserID` because there is no real need for it: the id is generated and saved right during the first install before any YouTube video is loaded (aka before the content script is injected and can try to request it).